### ISSUE TITLE
Adding support for .no TLD (norid)

### ISSUE THREAD

**Issue by feinom** [datahaven-net/epp-python-client#31]
Hi,

I am currently working on adding support for .no domains. They have several .no specific extensions that I need to add support for. Is there any specific way you would prefer that this is implemented? I think the most clean way of doing this would perhaps be to create new .no specific files in the commands directory instead of adding more to the existing files?

**Comment by vesellov** [comment]
Hi @feinom !

Without knowing details about those required extensions I can not say which way is the best to implement.

You can indeed create a new file `extension.py` because those are also part of EPP RFC: https://datatracker.ietf.org/doc/html/rfc3735

I suspect we also have to modify the other files in the `commands\` sub folder to be able to inject extensions in the EPP payloads.

Would be really great to have this implemented backward-compatible way. Please let me know how I can assist you further.

**Comment by feinom** [comment]
I discovered that the order of some of the items in the XML has to be rearranged to make it work with NORID as well.  Would it be easier if I created a subdirectory/module in the commands directory that contains all of the NORID/.no specific XML? That way I wouldn't have to touch any of the existing commands, and ensure backwards compatibility.

**Comment by vesellov** [comment]
yes, but then we will have a precedent for similar cases. and eventually we will have all kind of different pieces of code related to different registries.

can you share an example where the order of XML tags matters ?
I believe we should be able to make it working backward compatible way as long as we are following the RFC. And I do not remember anything specific was mentioned about order of the tags....

You can always start a pull request and there we can continue the discussion.

Another option is to create a separate branch in the repository where you implement specific changes for .no TLD. Later we can study, test and find a way to merge things together.
